File: src/main/java/Library/Recording.java
package Library;

import java.time.LocalDate;

public class Recording {
    private final Book book;
    private final LocalDate dueDate;

    public Recording(Book book, LocalDate dueDate){
        this.book = book;
        this.dueDate = dueDate;
    }

    // getters
    public Book getBook(){
        return null;
    }
    public LocalDate getDueDate(){
        return null;
    }

}

File: A3/models/Recording.js
// Recording.js

class Recording {
    constructor(book, dueDate) {
        this.book = book;        // keep direct reference
        this.dueDate = dueDate;  // JS Date object or null
    }

    // getters
    getBook() {
        return this.book;
    }

    getDueDate() {
        return this.dueDate;
    }
}

module.exports = Recording;
